sci_strategic_plugin/src/sci_strategic_plugin.cpp

```diff
@@ -230,17 +230,19 @@ bool SCIStrategicPlugin::planManeuverCb(cav_srvs::PlanManeuversRequest& req, cav
 
   double distance_to_stopline = stop_intersection_down_track - current_state.downtrack;
 
+  // Identify the lanelets which will be crossed by approach maneuvers lane follow maneuver
+      std::vector<lanelet::ConstLanelet> crossed_lanelets =
+          getLaneletsBetweenWithException(current_state.downtrack, stop_intersection_down_track, true, true);
+
+
   if (distance_to_stopline >= 0)
   {
     if (distance_to_stopline > config_.stop_line_buffer)
     {
       // approaching stop line
       
 
-      // Identify the lanelets which will be crossed by approach maneuvers lane follow maneuver
-      std::vector<lanelet::ConstLanelet> crossed_lanelets =
-          getLaneletsBetweenWithException(current_state.downtrack, stop_intersection_down_track, true, true);
-
+      
       speed_limit_ = findSpeedLimit(crossed_lanelets.front());
```

Comment:
Is there a chance we are in the lanelet preceding the stop controlled intersection?
Should we be using the speed limit of the current lanelet?